### PR TITLE
Updates html pipeline with dev vs production logic for js files

### PR DESCRIPTION
**Pull request by mattlaguardia**
This PR addresses this issue where ext js file aren't loading when they should in dev. See thread for more details (https://imgix.slack.com/archives/C02FLP0B23C/p1680125391811699?thread_ts=1679943660.222989&cid=C02FLP0B23C). The fix is to wrap this logic in a condition for prod environment.

### CODE REVIEW

**Review by luqven — APPROVED** [review]
LGTM 🚀 



Curious what the ELI5 is for why `async defer` throws off the external modules in dev but not prod 🤔 .

**Comment by mattlaguardia** [comment]
My guess is that due to those js being a larger bundle sizes they load after the small js files that need them (this is only on dev) in prod we bundle the JS better.
